travis: Avoid unnecessarily setting env variables on the lint build

Empact opened the pull request:
The relevent env variables are set for the matrix builds, and are irrelevant
to the lint build. By default the first matrix entry is applied.

"Each job included in jobs.include inherits the first value of the array that defines a matrix dimension."
https://docs.travis-ci.com/user/build-stages/#Build-Stages-and-Build-Matrix-Expansion

Note the global env variables are still applied, just the matrix are excluded:
https://travis-ci.org/bitcoin/bitcoin/jobs/406455565#L558

Before:
<img width="755" alt="screen shot 2018-07-20 at 19 16 45" src="https://user-images.githubusercontent.com/5470/43029041-81e5fd14-8c51-11e8-9e2a-6c4bcbb36d2c.png">

After:
<img width="702" alt="screen shot 2018-07-20 at 19 23 05" src="https://user-images.githubusercontent.com/5470/43029187-5ec76b28-8c52-11e8-8b03-5bee859c0d96.png">

Empact committed:
Avoid unnecessarily setting env variables on the lint build

The relevent env variables are set for the matrix builds, and are irrelevant
to the lint build. By default the first matrix entry is applied.

"Each job included in jobs.include inherits the first value of the array that defines a matrix dimension."
https://docs.travis-ci.com/user/build-stages/#Build-Stages-and-Build-Matrix-Expansion

MarcoFalke commented:
Since the global vars are still exported, this doen't really matter, imo. But utACK, does no harm either.

MarcoFalke commented on .travis.yml:
```diff
@@ -81,6 +81,7 @@ after_script:
+      env:
       sudo: false
       cache: false
       addons:
```
As you touch this file, might as well do some other cleanup and e.g. remove the apt addon stage completely. The installation is not needed (and only consumes time). See diff, that I confirmed to be working:

```diff
diff --git a/.travis.yml b/.travis.yml
index ec78896f44..c03a2bea24 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -82,13 +82,10 @@ jobs:
     - stage: lint
       sudo: false
       cache: false
-      addons:
-        apt:
-          packages:
-            - python3-pip
-            - shellcheck
+      language: python
+      python: '3.6'
       install:
-        - travis_retry pip3 install flake8 --user
+        - travis_retry pip install flake8
       before_script:
         - git fetch --unshallow
       script:
```

Empact replied:
Nice, 26 seconds saved: [before](https://travis-ci.org/bitcoin/bitcoin/builds/406455557?utm_source=github_status&utm_medium=notification) [after](https://travis-ci.org/bitcoin/bitcoin/builds/406632797?utm_source=github_status&utm_medium=notification)

fanquake commented:
Concept ACK. Additional cleanups also look ok.

Empact committed:
Don't unnecessarily install shellcheck on the lint build

"Travis CI has now integrated ShellCheck by default, so you don't need to manually install it."
https://github.com/koalaman/shellcheck#travis-ci

Empact committed:
Use the Travis python language feature on the lint build

Rather than installing python via apt.

https://docs.travis-ci.com/user/languages/python/

MarcoFalke commented:
utACK 751c9587d42b4c603ef85e5e846a96b2e9dd7590

fanquake commented:
utACK 751c958
